pkg/operator/admission/validation/pullsecret/validator_test.go

```diff
@@ -0,0 +1,246 @@
+			statusCode:  -1,
+			returnedErr: fmt.Errorf("some error"),
+			wantErr:     "some error",
+		},
+		//headers are not set, but tested when tested extractValuesFromAuthHeader
+		{
+			name:       "200",
+			statusCode: 200,
+			wantErr:    "header is missing data",
+		},
+		//headers are not set, but tested when tested extractValuesFromAuthHeader
+		{
+			name:       "401",
+			statusCode: 401,
+			wantErr:    "header is missing data",
+		},
+	} {
+		t.Run(tt.name, func(*testing.T) {
+			log := logrus.New()
+			log.SetOutput(ioutil.Discard)
+			entry := logrus.NewEntry(log)
+			mockController := gomock.NewController(t)
+			defer mockController.Finish()
+			requestDoerMock := mock_pullsecretadmission.NewMockRequestDoer(mockController)
+
+			requestDoerMock.
+				EXPECT().
+				Do(gomock.Any()).
+				AnyTimes().
+				Return(
+					&http.Response{
+						StatusCode: tt.statusCode,
+					},
+					tt.returnedErr)
+
+			client := ociRegClient{httpClient: requestDoerMock, ctx: context.Background()}
+			_, _, err := client.fetchAuthURL(entry, "")
+			if err.Error() != tt.wantErr {
+				t.Errorf("fetchAuthURL failed, wanted err=%s but got err=%s", tt.wantErr, err)
+			}
+		})
+	}
+}
+
+func TestGetToken(t *testing.T) {
+	for _, tt := range []struct {
+		name        string
+		statusCode  int
+		returnedErr error
+		wantErr     string
+		body        io.Reader
+		wantToken   string
+	}{
+		{
+			name:       "wrong status",
+			statusCode: 500,
+			wantErr:    fmt.Sprintf("authentication unsucessful, got status code %d but wanted %d", 500, http.StatusOK),
+		},
+		{
+			name:        "response error",
+			statusCode:  500,
+			returnedErr: fmt.Errorf("some error"),
+			wantErr:     "some error",
+		},
+		{
+			name:       "success",
+			statusCode: 200,
+			body:       strings.NewReader(`{"Token":"supertoken"}`),
+			wantToken:  "supertoken",
+		},
+		{
+			name:       "wrong json",
+			statusCode: 200,
+			body:       strings.NewReader(`{"wrongjson":"notcorrect"}`),
+			wantToken:  "",
+			wantErr:    "no token in response",
+		},
+	} {
+		t.Run(tt.name, func(*testing.T) {
+			log := logrus.New()
+			log.SetOutput(ioutil.Discard)
+			entry := logrus.NewEntry(log)
+			mockController := gomock.NewController(t)
+			defer mockController.Finish()
+			requestDoerMock := mock_pullsecretadmission.NewMockRequestDoer(mockController)
+
+			requestDoerMock.
+				EXPECT().
+				Do(gomock.Any()).
+				AnyTimes().
+				Return(
+					&http.Response{
+						StatusCode: tt.statusCode,
+						Body:       io.NopCloser(tt.body),
+					},
+					tt.returnedErr)
+
+			client := ociRegClient{httpClient: requestDoerMock, ctx: context.Background()}
+			token, err := client.getToken(entry, "", "", "user", "password")
+			if err == nil && tt.wantErr != "" {
+				t.Errorf("gettoken failed, wanted err=%s but got err=%s", tt.wantErr, err)
+			}
+			if err != nil && err.Error() != tt.wantErr {
+				t.Errorf("gettoken failed, wanted err=%s but got err=%s", tt.wantErr, err)
+			}
```

Comment:
We could use an "or" to simplify it a bit.

Reply:
same as the others, when it fails it will say at what line, and make it easier to know what part of the test failed